modules/@apostrophecms/attachment/index.js

```diff
@@ -454,6 +462,21 @@ module.exports = {
         await self.db.insertOne(info);
         return info;
       },
+
+      async update(req, file, attachment) {
+        const existing = await self.db.findOne({ _id: attachment._id });
+        if (!existing) {
+          throw self.apos.error('notfound');
+        }
+        await self.alterAttachment(existing, 'remove');
+        await self.db.deleteOne({ _id: existing._id });
+        await self.insert(req, file, {
+          attachmentId: attachment._id,
+          docIds: attachment.docIds,
```

Comment:
Merge the old and new docIds and archivedDocIds so you have a complete list. Conversely, use `$in` queries with an `{ _id: 1 }` projection to confirm they all actually exist on the receiving site and filter out those that don't. (I assume you're doing all of the attachments last after the docs so that this would give a complete result.)

Reply:
I know we are pressed for time. There is also a much simpler alternative re: tracking docIds and archivedDocIds. Just use the `recomputeAllDocReferences` task programmatically at the end of the entire import process. It takes a little time, but and it will definitely get the right result without any tricky bookkeeping. It is 100% OK to do that for this first release at least, if you find it difficult to get the lists right per the technique above.

You can invoke a task internally but I'd just refactor its logic to a method for convenience (with no other changes).